packages/graph/hash_graph/lib/graph/src/store/postgres/mod.rs

```diff
@@ -70,34 +80,115 @@ impl<K> Default for DependencyMap<K> {
+                    // new:  [---)
                     DependencyStatus::Resolved
+                } else if new_interval.contains_interval(&old_interval)
+                    || new_interval.is_adjacent_to(&old_interval)
+                {
+                    // The dependency is already resolved, but not for the required interval or the
+                    // intervals are adjacent so we only have to resolve the new interval
+                    // old:  [---)   |  [---)
+                    // new: [-----)  |      [---)
+                    DependencyStatus::Unresolved(*current_depths, new_interval)
+                } else if old_interval.overlaps(&new_interval) {
+                    // The time intervals either overlap or are disjoint, so only have to resolve
+                    // the difference of the two intervals.
+                    // old:     [-----)
+                    // new:       [-------)
+                    // resolve:       [---)
+                    let difference = new_interval.difference(old_interval);
+                    // The intervals do overlap and the current interval is not a subset of
+                    // the required interval, so the difference must be a single interval
+                    debug_assert_eq!(difference.len(), 1, "difference must be a single interval");
+
+                    DependencyStatus::Unresolved(
+                        *current_depths,
+                        difference
+                            .into_iter()
+                            .next()
+                            .expect("difference must be a single interval"),
+                    )
+                } else {
+                    // The time intervals are disjoint. Ideally, we only would resolve the new
+                    // interval, but we did not come up with a good way to store the different
+                    // intervals in the dependency map. So we resolve the full interval for now.
+
+                    // This case is expected to happen very rarely, so falling back to this logic
+                    // is fine for now. If it turns out to be a performance bottleneck, we can
+                    // find and implement a better solution. To track this, we log a warning.
+                    tracing::warn!(
+                        vertex_id=?identifier,
+                        ?old_interval,
+                        ?new_interval,
+                        "dependency is resolved for disjoint intervals"
+                    );
+
+                    // We only require this logic when traversing edges of the graph, so the roots
+                    // of the graph are always precisely resolved correctly. By this implementation,
+                    // we may get a few more dependencies resolved than necessary, which will appear
+                    // as vertices/edges in the subgraph. As we do not guarantee that the subgraph's
+                    // vertices/edges are minimal, this is not a problem.
```

Comment:
I disagree with this, and I thought we explicitly talked about this in a call. I _do_ view this as a problem. If we want to accept this as a "feature not a bug" then I think we need to talk about the ramifications for this, I can easily see a lot of block developers creating UIs to display the "number of friends of a person" for example, which would be based on the elements of the edge set. If you cannot rely on that, then I do not know what you can correctly rely on. This seems like a foundational essential part of having a temporal datastore.

Reply:
We've spoken about this more on a call. We're returning too much information in `main` at the moment anyway, so we will probably address this in a follow-up if we feel it is worth addressing (as this PR shouldn't be a regression).